Fix complex tensor printing

anjali411 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#38031 Fix complex tensor printing**

Differential Revision: [D21502915](https://our.internmc.facebook.com/intern/diff/D21502915)

anjali411 committed:
Fix tensor printing

[ghstack-poisoned]

anjali411 committed:
Update on "Fix tensor printing"




[ghstack-poisoned]

SsnL commented on torch/_tensor_str.py:
```diff
@@ -280,7 +280,9 @@ def _str(self):
 
-    has_default_dtype = self.dtype in (torch.get_default_dtype(), torch.int64, torch.bool)
+    # TODO: add an API to map real -> complex dtypes
+    _default_complex_dtype = torch.cfloat if torch.get_default_dtype() == torch.float else torch.cdouble
```
so if default dtype is half this is cdouble?

anjali411 replied:
good point! no it should be cdouble if double else cfloat, similar to the logic [here](https://github.com/pytorch/pytorch/blob/a44824c9eda0301bb497840904ac1c64f0c69b89/c10/core/DefaultDtype.cpp#L11)

anjali411 replied:
will update!

anjali411 committed:
Update on "Fix tensor printing"




[ghstack-poisoned]

ezyang reviewed (approved):
Test please? :)

anjali411 committed:
Update on "Fix complex tensor printing"


Differential Revision: [D21502915](https://our.internmc.facebook.com/intern/diff/D21502915)

[ghstack-poisoned]
